lib/Doctrine/ORM/Mapping/Driver/YamlDriver.php

```diff
@@ -45,6 +47,14 @@ public function __construct($locator, $fileExtension = self::DEFAULT_FILE_EXTENS
             'YAML mapping driver is deprecated and will be removed in Doctrine ORM 3.0, please migrate to attribute or XML driver.'
         );
 
+        if (! class_exists(Yaml::class)) {
```

Comment:
Let's use https://github.com/composer/composer/blob/3ae111140facdba8ae82adcd1085e4adfc7d715c/src/Composer/InstalledVersions.php#L95 ?

Reply:
Can you elaborate what we would gain from avoiding `class_exists()` here?

Reply:
It's very hypothetical of course, but in case there is a gigantic BC break and the `Yaml` class no longer exists in this package, it would be confusing for the user if they still have the package installed but get a message letting them know it is not installed.

Reply:
The error that they would get if the `InstalledVersions` check would pass although that class is gone would be even more cryptic and confusing. Also, apart from `symfony/yaml`, this class could also be provided by `symfony/symfony` (although depending on the monorepo is discouraged).

I think, I'd stick to the `class_exists()` call if you don't mind. This is the class we're interacting with, so checking for its existence feels like the right thing to do.

Reply:
> The error that they would get if the InstalledVersions check would pass although that class is gone would be even more cryptic and confusing.

The error they would get would be that `Yaml` doe snot exist… it's not cryptic nor confusing IMO, and gives them a better thing to google.

Good point about `symfony/symfony` though, let's stick with `class_exists`